source/common/http/conn_manager_impl.cc

```diff
@@ -504,6 +504,8 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
     is_head_request_ = true;
   }
 
+  maybeEndDecode(end_stream);
```

Comment:
It may be worth landing this and the regression test as their own small standalone PR.  I wonder if it's the cause of the integration test failures?

Reply:
Yup, I'll do this in a separate PR.